src/config_attribute_value.hpp

```diff
@@ -140,7 +140,7 @@ class config_attribute_value
 	config_attribute_value& operator=(const t_string &v);
 
 	/** Calls @ref operator=(const std::string&) if @a v is not empty. */
-	void write_if_not_empty(const std::string& v);
+	void write_if_not_empty(const t_string& v);
```

Comment:
I mean, I guess it works, but… the function is only 3 lines so I don't really see such a problem with just duplicating the same 3 lines, which will mean that it ends up stored as a string instead of a t_string in the variant… though I don't know if there's a real advantage to that or not. I suppose if there's an advantage it might be related to memory, since a t_string just is a pointer, whereas a sufficiently short string is stored inline.